tests/notifications/test-class-yoast-notification.php

```diff
@@ -0,0 +1,322 @@
+		$filter_function = array( $this, 'add_wpseo_notification_capabilities' );
+
+		$me = wp_get_current_user();
+
+		$capabilities = $this->add_wpseo_notification_capabilities();
+		foreach ( $capabilities as $capability ) {
+			$me->add_cap( $capability );
+		}
+
+		$me->remove_cap( 'manage_options' );
+
+		$subject = new Yoast_Notification( 'message', array( 'id' => 'id', 'capabilities' => 'manage_options' ) );
+
+		$this->assertFalse( $subject->display_for_current_user() );
+
+		add_filter( 'wpseo_notification_capabilities', $filter_function );
+		$this->assertTrue( $subject->display_for_current_user() );
+		remove_filter( 'wpseo_notification_capabilities', $filter_function );
+	}
+
+	/**
+	 * Test capability filter
+	 */
+	public function test_wpseo_notification_capability_match() {
+		$filter_function = array( $this, 'add_wpseo_notification_capabilities' );
+		$match_function  = array( $this, 'add_wpseo_notification_capability_check' );
+
+		// Only add 1 of the provided capabilities. We need 'any' to match this.
+		$capabilities = $this->add_wpseo_notification_capabilities();
+
+		$me = wp_get_current_user();
+		$me->add_cap( $capabilities[0] );
+		$me->remove_cap( $capabilities[1] );
+
+		$me->remove_cap( 'manage_options' );
+
+		$subject = new Yoast_Notification( 'message', array( 'id' => 'id', 'capabilities' => array( 'manage_options' ) ) );
+
+		$this->assertFalse( $subject->display_for_current_user() );
+
+		add_filter( 'wpseo_notification_capability_check', $match_function );
+		add_filter( 'wpseo_notification_capabilities', $filter_function );
+		$this->assertTrue( $subject->display_for_current_user() );
+		remove_filter( 'wpseo_notification_capabilities', $filter_function );
+		remove_filter( 'wpseo_notification_capability_check', $match_function );
+	}
+
+	/**
+	 * Invalid filter return value
+	 */
+	public function test_invalid_filter_return_values() {
+		$match_function = array( $this, 'empty_output' );
+
+		$subject = new Yoast_Notification( 'message', array( 'id' => 'id', 'capabilities' => array( 'xijwe' ) ) );
+
+		add_filter( 'wpseo_notification_capability_check', $match_function );
+		add_filter( 'wpseo_notification_capabilities', $match_function );
+		$this->assertTrue( $subject->display_for_current_user() );
+		remove_filter( 'wpseo_notification_capabilities', $match_function );
+		remove_filter( 'wpseo_notification_capability_check', $match_function );
+	}
+
+	/**
+	 * Test notification filter
+	 *
+	 * @param array $input Input.
+	 *
+	 * @return array
+	 */
+	public function add_wpseo_notification_capabilities( $input = array() ) {
+		return array( 'jip', 'janneke' );
+	}
+
+	/**
+	 * Any
+	 *
+	 * @param array $input Input.
+	 *
+	 * @return string
+	 */
+	public function add_wpseo_notification_capability_check( $input = array() ) {
+		return 'any';
+	}
+
+	/**
+	 * Return invalid output
+	 *
+	 * @param null $input Input.
+	 *
+	 * @return string
+	 */
+	public function empty_output( $input = null ) {
```

Comment:
If you ever need a function that returns an empty string for a filter you can use [`__return_empty_string`](https://developer.wordpress.org/reference/functions/__return_empty_string/)